crates/router/src/connector/powertranz/transformers.rs

```diff
@@ -29,6 +29,7 @@ pub struct PowertranzPaymentsRequest {
     // billing_address: Option<PowertranzAddressDetails>,
     // shipping_address: Option<PowertranzAddressDetails>,
     extended_data: Option<ExtendedData>,
+    reference: String,
```

Comment:
Hi @kritikash18 ,

In reference to issue #2345 , we need to adjust the reference field mapping based on the API documentation of the connector. `order_identifier` is merchant reference id for powertranz. 

the api documentation for powertranz is attached below
[PowertranzNon3DSAPI-V-2.0.pdf](https://github.com/juspay/hyperswitch/files/12789748/PowertranzNon3DSAPI-V-2.0.pdf)

Reply:
updated

Reply:
Currently `order_identifier` is mapped to `payment_id` , it should be updated to `connector_request_reference_id` in payment request try_from .

Reply:
updated; I'm new to rust so apologies for the extra review effort 😅